components/editor/editor/components/bubble-menu.tsx

```diff
@@ -23,34 +25,56 @@ type EditorBubbleMenuProps = Omit<Required<Pick<BubbleMenuProps, "editor">> & Bu
 
 export const EditorBubbleMenu: FC<EditorBubbleMenuProps> = (props) => {
   const items: BubbleMenuItem[] = [
+    {
+      name: "heading",
+      isActive: () => props.editor?.isActive("heading", { level: 2 }) ?? false,
+      command: () => props.editor?.chain().focus().toggleHeading({ level: 2 }).run(),
+      icon: Heading,
+    },
+    {
+      name: "text",
+      isActive: () => props.editor?.isActive("paragraph") ?? false,
+      command: () => props.editor?.chain().focus().setParagraph().run(),
+      icon: TextIcon,
+    },
     {
       name: "bold",
-      isActive: () => props.editor.isActive("bold"),
-      command: () => props.editor.chain().focus().toggleBold().run(),
+      isActive: () => props.editor?.isActive("bold") ?? false,
+      command: () => props.editor?.chain().focus().toggleBold().run(),
```

Comment:
I've updated the error handling in `bubble-menu.tsx` to use optional chaining and null coalescing as suggested, since TypeScript was indicating that `props.editor` could potentially be null. This change provides better type safety while maintaining the component's functionality.